### ISSUE TITLE
torch.max does not reduce correctly dimension of size 1

### ISSUE THREAD

**Issue by aosokin** [pytorch/pytorch#2317]
Hi, on recent release 0.2, this code
```
import torch
a = torch.LongTensor(5, 2)
print(a.size(), a.max(0, keepdim=False)[0].size())
b = torch.LongTensor(1, 2)
print(b.size(), b.max(0, keepdim=False)[0].size())
```
prints this
```
torch.Size([5, 2]) torch.Size([2])
torch.Size([1, 2]) torch.Size([1, 2])
```
Is it the expected behavior or a bug? Should the last statement be `torch.Size([2])`?
torch.min behaves the same way; torch.sum is different and eliminates the dimension

**Comment by fmassa** [comment]
This looks like a bug to me.
I think the problem comes from [this line](https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/generic/THTensorMath.c#L1597), which has a special handling case for `dim==1`.
A solution would be to copy [this part](https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/generic/THTensorMath.c#L1621-L1624) in that condition for `dim==1`.
Could you send a PR fixing that?

**Comment by aosokin** [comment]
@fmassa thanks, PR #2318 is on its way
